src/game/client/tf/tf_hud_speedo.cpp

```diff
@@ -0,0 +1,217 @@
+//========= Contributed by 4v4 PASS Time developers. ==========================//
+//
+// Purpose: Shows player speed on the HUD.
+// 
+//
+// $NoKeywords: $
+//=============================================================================//
+
+#include "cbase.h"
+#include "usermessages.h"
+#include "hud.h"
+#include "hudelement.h"
+#include "c_tf_player.h"
+#include "iclientmode.h"
+#include "ienginevgui.h"
+#include "tf_gamerules.h"
+#include "hud_macros.h"
+#include <vgui/ILocalize.h>
+#include <vgui/ISurface.h>
+#include <vgui/IVGui.h>
+#include <vgui_controls/EditablePanel.h>
+#include <vgui_controls/Label.h>
+#include "tf_shareddefs.h"
+#include <vgui_controls/ImagePanel.h>
+
+// memdbgon must be the last include file in a .cpp file!!!
+#include "tier0/memdbgon.h"
+
+using namespace vgui;
+
+ConVar pf_speedo( "pf_speedo", "1", FCVAR_ARCHIVE, "Enable/Disable the speedometer HUD element" );
+ConVar pf_speedo_smoothing( "pf_speedo_smoothing", "0", FCVAR_ARCHIVE, "Enable/disable smoothing for the HUD speedometer." );
```

Comment:
I think smoothing looks _significantly_ better and should be enabled by default. Would be visually more appealing for most players and new players especially.